bftengine/src/bftengine/ViewsManager.hpp

```diff
@@ -134,6 +136,19 @@ class ViewsManager {
 
   bool hasViewChangeMessageForFutureView(uint16_t repId);
 
+  ReplicasAskedToLeaveViewInfo &getComplainedReplicas() { return complainedReplicas; }
```

Comment:
Instead of exposing the `complainedReplicas` object, can we instead expose several methods in `ViewsManager` with the supported operations:
* `getComplaintFromReplica()`
* `addComplaint()`
* `hasQuorumToLeaveView()`
* ...

@HristoStaykov what are your thoughts on such an approach?

Reply:
I agree. The only place where we need to iterate over all complaints is during Status Exchange to see what the Peer requesting Status Exchange is missing, compared to what we have and help accordingly. [ReplicaImp.cpp:2220](https://github.com/vmware/concord-bft/blob/c197dc3b8703e0b859d847e0dbb24248eafb4c55/bftengine/src/bftengine/ReplicaImp.cpp#L2220). The filling of this information to Status Request messages I see is correctly moved to ViewsManager.